File: unimonpark/src/main/java/co/edu/unimonserrate/unimonpark/entity/Factura.java
package co.edu.unimonserrate.unimonpark.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import co.edu.unimonserrate.unimonpark.enums.EstadoFactura;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "facturas")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Factura {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idFactura;
    
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_salida", nullable = false, unique = true)
    private Salida salida;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_usuario", nullable = false)
    private Usuario usuario;

    @Column(nullable = false)
    private LocalDateTime fecha;

    @Column(nullable = false)
    private BigDecimal subtotal;

    @Column(nullable = false)
    private BigDecimal descuento;

    @Column(nullable = false)
    private BigDecimal iva;

    @Column(nullable = false)
    private BigDecimal total;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EstadoFactura estado;
}

File: unimonpark/unimonpark-frontend/src/types/factura.ts
export interface Factura {
    idFactura: number;
    idUsuario?: number | null;
    idExterno?: number | null;
    nombres: string;
    apellidos: string,
    fecha: string;
    subtotal: number;
    descuento: number | null;
    iva: number | null;
    total: number;
    estado: string;
    idSalida: number;
    idTarifa: string;
    nombreTarifa: string;
    placaVehiculo: string;
    tipoVehiculo: string;
    categoriaPersona: string;
}

export interface FacturaPayload {
    idUsuario?: number | null;
    idExterno?: number | null;
    descuento: number;
    iva: number;
    idSalida: number;
}
